int64 speed1
int64 speed2